src/sagemaker/lineage/query.py

```diff
@@ -259,23 +285,182 @@ def _get_app(self, elements):
                                 "label": "data(label)",
                                 "color": "gray",
                                 "text-halign": "left",
-                                "text-margin-y": "3px",
-                                "text-margin-x": "-2px",
-                                "font-size": "3%",
-                                "width": "1%",
-                                "curve-style": "taxi",
+                                "text-margin-y": "2.5",
+                                "font-size": "3",
+                                "width": "1",
+                                "curve-style": "bezier",
+                                "control-point-step-size": "15",
                                 "target-arrow-color": "gray",
                                 "target-arrow-shape": "triangle",
                                 "line-color": "gray",
                                 "arrow-scale": "0.5",
+                                "font-family": "verdana",
                             },
                         },
+                        {"selector": ".Artifact", "style": {"background-color": "#146eb4"}},
+                        {"selector": ".Context", "style": {"background-color": "#ff9900"}},
+                        {"selector": ".TrialComponent", "style": {"background-color": "#f6cf61"}},
+                        {"selector": ".Action", "style": {"background-color": "#88c396"}},
+                        {"selector": ".startarn", "style": {"shape": "star"}},
+                        {"selector": ".select", "style": {"border-opacity": "0.7"}},
                     ],
                     responsive=True,
-                )
+                ),
+                self.html.Div(
+                    style={
+                        "width": "0.5%",
+                        "display": "inline-block",
+                        "font-size": "1vw",
+                        "font-family": "verdana",
+                        "vertical-align": "top",
+                    },
+                ),
+                self.html.Div(
+                    [
+                        self.html.Div(
+                            [
```

Comment:
It looks like most of this is the same for the different nodes, can we convert this into a function that takes the title (ex: Trial Component") and background color as parameters? Then we can just call that function for each of the different types

Are these the divs for the legend?  Can you add some more comments explaining in the code what each section is for? (for example: explaining that this section is the legend) Many people (myself included!) might be new to dash so I think comments could be helpful

Reply:
Created a function `self._create_legend_component()` that returns html.div components for the legend section. Also created a dictionary `self.entity_color` that stores lineage entity as key and color as value. We traverse through the dictionary and call `self._create_legend_component()` to get all of the legend components when rendering the graph

Reply:
Nice, thanks